orchagent/muxorch.cpp

```diff
@@ -1283,14 +1283,13 @@ void MuxOrch::updateRoute(const IpPrefix &pfx, bool add)
     for (auto it = nextHops.begin(); it != nextHops.end(); it++)
     {
         NextHopKey nexthop = *it;
-        /* This will only work for configured MUX neighbors (most cases)
-         * TODO: add way to find MUX from neighbor
-         */
-        MuxCable* cable = findMuxCableInSubnet(nexthop.ip_address);
-        auto standalone = standalone_tunnel_neighbors_.find(nexthop.ip_address);
+        NeighborEntry neighbor;
+        MacAddress mac;
+
+        gNeighOrch->getNeighborEntry(nexthop, neighbor, mac);
+        auto standalone = standalone_tunnel_neighbors_.find(neighbor.ip_address);
 
-        if ((cable == nullptr && standalone == standalone_tunnel_neighbors_.end()) ||
-             cable->isActive())
+        if (isNeighborActive(neighbor.ip_address, mac, neighbor.alias) && standalone == standalone_tunnel_neighbors_.end())
```

Comment:
This feels redundant to me, if a neighbor has a standalone tunnel route that means it has a zero MAC entry and isNeighborActive should return false.